src/server/modes/charts/plugins/ql/js.ts

```diff
@@ -246,6 +252,24 @@ export default ({shared, ChartEditor}: {shared: QlConfig; ChartEditor: IChartEdi
             });
         }
 
+        if (newVisualization.placeholders[0]?.items[0]) {
+            if (
+                !applyDefaultSorting &&
+                (LINEAR_VISUALIZATIONS.has(newVisualization.id) ||
+                    newVisualization.id === WizardVisualizationId.BarXD3)
+            ) {
+                newVisualization.placeholders[0].settings = {
+                    axisModeMap: {
+                        [newVisualization.placeholders[0].items[0].guid]: 'discrete',
+                    },
+                };
+
+                newVisualization.placeholders[0].items[0].disableAxisMode = true;
+            } else {
```

Comment:
But in general, I don't think it is necessary to store a flag for the field disabling. It would be more useful to keep the knowledge about the sorting in the query (may be somewhere in metadata?). In this case, the field disable flag can be calculated, and the knowledge about sorting can be used somewhere else.